relay-server/src/actors/spooler/mod.rs

```diff
@@ -490,6 +527,60 @@ impl OnDisk {
         }
     }
 
+    /// Unspools the entire contents of the on-disk spool.
+    async fn delete_and_fetch_all(
+        &mut self,
+        services: &Services,
+    ) -> Result<BTreeMap<QueueKey, Vec<ManagedEnvelope>>, BufferError> {
+        let mut result: BTreeMap<QueueKey, Vec<ManagedEnvelope>> = BTreeMap::new();
+
+        loop {
+            let mut envelopes =
+                sql::delete_and_fetch_all(BATCH_SIZE.min(self.buffer_guard.available() as i64))
+                    .fetch(&self.db)
+                    .peekable();
+            relay_statsd::metric!(counter(RelayCounters::BufferReads) += 1);
+
+            // Stream is empty, we can break the loop, since we read everything by now.
+            if Pin::new(&mut envelopes).peek().await.is_none() {
+                break;
+            }
+
+            let mut count: i64 = 0;
+            while let Some(envelope) = envelopes.next().await {
+                count += 1;
+                let envelope = match envelope {
+                    Ok(envelope) => envelope,
+
+                    // Bail if there are errors in the stream.
+                    Err(err) => {
+                        relay_log::error!(
+                            error = &err as &dyn Error,
+                            "failed to read the buffer stream from the disk",
+                        );
+                        self.track_count(-count);
```

Comment:
How does this work here, if the stream contains an error:

* does it end or can we just skip over the error?
* can there be more rows deleted than returned?

The count may not be correct here then, also we maybe shouldn't stop on error but just skip over them?

Reply:
IIUC, not skipping a stream error may result in losing all envelopes?

Reply:
skipping the error now and trying to read out all the envelopes instead , done in [`b2b35f2`](https://github.com/getsentry/relay/pull/2949/commits/b2b35f2b50f796378f0b84da90c7061eaa9474b6)